### PR TITLE
Fix longformer attention mask type casting when using apex

### PR DESCRIPTION
**Pull request by wfangtw**
Fix for issue [#4525](https://github.com/huggingface/transformers/issues/4525).

### CODE REVIEW

File: src/transformers/modeling_longformer.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def forward(
             ]
             attn[extra_attention_mask_nonzeros[::-1]] = nonzero_selected_attn.view(
                 len(selection_padding_mask_nonzeros[0]), -1
-            ).type_as(hidden_states)
+            ).type_as(attn)
```
Comment (ibeltagy) [inline src/transformers/modeling_longformer.py:376]: Thanks,  @peskotivesgeroff. I think `type_as` is not needed at all. Do you mind removing it and see if it works? I think `type_as` in line 311 is also not needed. Both were useful when we were [running the tvm code in fp32](https://github.com/allenai/longformer/blob/master/longformer/longformer.py#L148) but I don't think they are needed here.

**Review by ibeltagy — CHANGES_REQUESTED** [review]

File: src/transformers/modeling_longformer.py (hunk 1 above)
Comment (peskotivesgeroff) [reply src/transformers/modeling_longformer.py]: Thanks for the suggestion. Removing both works with apex amp, and I've updated the PR.

**Review by LysandreJik — APPROVED** [review]
